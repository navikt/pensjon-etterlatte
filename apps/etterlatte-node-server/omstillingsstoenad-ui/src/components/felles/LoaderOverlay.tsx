import { BodyLong, Loader } from '@navikt/ds-react'
import styled from 'styled-components'

const SpinnerOverlay = styled.div`
    position: fixed;
    display: block;
    width: 100vw;
    height: 100vh;
    top: 0;
    left: 0;
    background-color: rgba(255, 255, 255, 0.75);
    z-index: 999;
    pointer-events: none;
    overflow: no-display;
`

const SpinnerContent = styled.div`
    padding: 3rem;
    background-color: white;
    border-radius: 5px;
    box-shadow: 5px 5px 15px 0 #ddd;
    position: fixed;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
`

const LoaderOverlay = ({ visible, label }: { visible: boolean; label: string }) => {
    if (!visible) return null

    return (
        <SpinnerOverlay>
            <SpinnerContent>
                <Loader />
                <BodyLong spacing>{label}</BodyLong>
            </SpinnerContent>
        </SpinnerOverlay>
    )
}

export default LoaderOverlay
