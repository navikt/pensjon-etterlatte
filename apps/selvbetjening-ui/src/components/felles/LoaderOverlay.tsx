import { BodyLong, Loader } from "@navikt/ds-react";

const LoaderOverlay = ({ visible, label }: {
    visible: boolean;
    label: string;
}) => {
    if (!visible) return null;

    return (
        <div className={"spinner-overlay"}>
            <div className={"spinner-content"}>
                <Loader />
                <BodyLong spacing>
                    {label}
                </BodyLong>
            </div>
        </div>
    );
};

export default LoaderOverlay;
