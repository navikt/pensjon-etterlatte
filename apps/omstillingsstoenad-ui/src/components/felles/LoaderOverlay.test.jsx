import { render } from '@testing-library/react'
import LoaderOverlay from './LoaderOverlay'
import { describe, expect, it } from 'vitest'

describe('LoaderOverlay', () => {
    it('skal vise loader', () => {
        const { getByText } = render(
                <LoaderOverlay visible={true} label="Testlabel" />)

        expect(getByText('Testlabel')).toBeDefined()
    })
    it('skal ikke vise loader', () => {
        const { container } = render(<LoaderOverlay label="Testlabel" />)
        expect(container.firstChild).toBeNull()
        expect(container).toMatchInlineSnapshot(`<div />`)
    })
})
