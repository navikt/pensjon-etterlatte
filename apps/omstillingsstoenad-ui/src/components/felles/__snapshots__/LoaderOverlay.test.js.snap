// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderOverlay skal ikke vise loader 1`] = `<div />`;
